src/components/PostDetails.tsx

```diff
@@ -1,106 +1,121 @@
-import React from 'react';
-import { Loader } from './Loader';
+import React, { useEffect } from 'react';
 import { NewCommentForm } from './NewCommentForm';
+import { Loader } from './Loader';
+import { Comment, CommentData } from '../types/Comment';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+type Props = {
+  isCommentLoading: boolean;
+  postComments: Comment[];
+  selectedPost: Post | null;
+  errorsComment: boolean;
+  onDeleteComment: (commentId: number) => void;
+  onAddComment: (newComment: CommentData) => void;
+  isButtonAddLoading: boolean;
+  onShowCommentFormChange: (item: boolean) => void;
+  isCommentFormVisible: boolean;
+  userSelected: User | null;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  isCommentLoading,
+  postComments,
+  selectedPost,
+  errorsComment,
+  onDeleteComment,
+  onAddComment,
+  isButtonAddLoading,
+  onShowCommentFormChange,
+  isCommentFormVisible,
+  userSelected,
+}) => {
+  useEffect(() => {
+    onShowCommentFormChange(false);
+  }, [selectedPost]);
```

Comment:
You are initializing your state with props in the `useEffect` hook. This can lead to bugs and it is generally not recommended. The state should be the minimal representation of the UI state and should not include props. Here are some links that explain this issue in detail: https://stackoverflow.com/a/50403930, https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state